internal/controller/apiextensions/composite/environment_selector.go

```diff
@@ -87,7 +88,7 @@ func (s *APIEnvironmentSelector) SelectEnvironment(ctx context.Context, cr resou
 				s.buildEnvironmentConfigRefFromRef(src.Ref),
 			)
 		case v1.EnvironmentSourceTypeSelector:
-			ec, err := s.lookUpConfigs(ctx, cr, src.Selector.MatchLabels)
+			ec, err := s.lookUpConfigs(ctx, cr, src.Selector.MatchLabels, rev.Spec.Environment.IsRequired())
```

Comment:
nit: what about doing the check outside of the lookupConfigs method? just to avoid the additional bool parameter